safe/new_impact_function/impact_function.py

```diff
@@ -0,0 +1,766 @@
+
+        :rtype: function
+
+        .. seealso:: console_progress_callback
+        """
+        return self._callback
+
+    @callback.setter
+    def callback(self, callback):
+        """Setter for callback property.
+
+        :param callback: A callback function reference that provides the
+            following signature:
+
+            progress_callback(current, maximum, message=None)
+
+        :type callback: function
+        """
+        self._callback = callback
+
+    def setup_impact_function(self):
+        """Automatically called when the hazard or exposure is changed.
+        """
+        if not self.hazard or not self.exposure:
+            return
+
+        # Set the algorithm
+        self.set_algorithm()
+
+        # Set the name
+        self._name = '%s %s on %s %s' % (
+            self.hazard.keyword.get('hazard').title(),
+            self.hazard.keyword.get('layer_geometry').title(),
+            self.exposure.keyword.get('exposure').title(),
+            self.exposure.keyword.get('layer_geometry').title(),
+        )
+
+        # Set the title
+        if self.exposure.keyword.get('exposure') == 'population':
+            self._title = tr('need evacuation')
+        else:
+            self._title = tr('be affected')
+
+    def set_algorithm(self):
+        if self.exposure.keyword.get('layer_geometry') == 'raster':
+            # Special case for Raster Earthquake hazard.
+            if self.hazard.keyword('hazard') == 'earthquake':
+                pass
+            else:
+                self.algorithm = self.raster_algorithm
+        elif self.exposure.keyword.get('layer_geometry') == 'point':
+            self.algorithm = self.point_algorithm
+        elif self.exposure.keyword.get('exposure') == 'structure':
+            self.algorithm = self.indivisible_polygon_algorithm
+        elif self.exposure.keyword.get('layer_geometry') == 'line':
+            self.algorithm = self.line_algorithm
+        else:
+            self.algorithm = self.polygon_algorithm
+
+    def preprocess(self):
+        """Run process before running the main work / algorithm"""
+        # Clipping
+        # Convert hazard to classified vector
+        # Aggregation if needed
+        pass
+
+    def run_algorithm(self):
+        """Run the algorithm
+        """
+        pass
+
+    def post_process(self):
+        """More process after getting the impact layer with data."""
+        # Post processor (gender, age, building type, etc)
+        # Notes, action
+
+        for post_processor in post_processors:
+            post_processor_output = self.run_single_post_processor(
+                post_processor)
+            self.set_state_process(
+                'post_processor',
+                'Post processor for %s %s.' % (
+                    post_processor['name'], post_processor_output))
+
+    def run(self):
+        self.preprocess()
+        self.run_algorithm()
+        self.post_process()
```

Comment:
I have added it. But comment it out since it will make the test failed because there is no impact layer created.